corehq/apps/users/models.py

```diff
@@ -1028,6 +1030,9 @@ def two_factor_disabled(self):
     def is_dimagi(self):
         return self.username.endswith('@dimagi.com')
 
+    def is_locked_out(self):
```

Comment:
Not blocking, but moving the `is_web_user` and toggle checks in here might be nice.

Reply:
I thought about it, couple of pretty random reasons I chose not to
- None of the `users/models` refer to toggles, so I didn't want to make that code be coupled with toggles.
- putting the checks in individual places made it so that the behaviour can be  controlled case by case. 

I will also probably just revert this after AKS audit.

Reply:
makes sense, thanks for explaining